backend/src/agents/workflow.agent.ts

```diff
@@ -31,34 +33,103 @@ export const runDistributionWorkflow = async (
   workflow: DistributionWorkflowModel
 ) => {
   const { source, destinations } = workflow;
-  let sourcePosts;
-
-  if (source.type == ContainerType.BOARD) {
-    sourcePosts = await dalPost.getByBoard(source.id);
-    sourcePosts = sourcePosts.map((p) => p.postID);
-  } else {
-    const bucket: BucketModel | null = await dalBucket.getById(source.id);
-    sourcePosts = bucket ? bucket.posts : [];
-  }
+  if (
```

Comment:
Can we split each distribution type's code into different smaller functions? i.e. have a function for `randomDistribution`, `tagDistribution`, etc..